### PR TITLE
Do not leak websockets on errors

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: src/js/components/devices/terminal.js [hunk 1]
```diff
@@ -136,12 +139,17 @@ export const Terminal = props => {
...
             setSnackbar('Error: ' + byteArrayToString(obj.body), 5000);
+            snackbarAlreadySet = true;
+            socket.close();
             onCancel();
           } else {
             setSessionId(obj.hdr.sid);
           }
         } else if (obj.hdr.typ === MessageTypeShell) {
           term.write(byteArrayToString(obj.body));
+        } else if (obj.hdr.typ === MessageTypeStop) {
```
Comment (alfrunes) [inline src/js/components/devices/terminal.js:150]: Maybe this should be a "catch all" (else) case?

**Review by alfrunes — APPROVED** [review]
I can indeed verify that it works. Well done!

File: src/js/components/devices/terminal.js (hunk 1 above)
Comment (tranchitella) [reply src/js/components/devices/terminal.js:150]: uhm, not really: we are at risk of closing the socket if we do not recognize a message...

Comment (tranchitella) [reply src/js/components/devices/terminal.js]: Maybe this one is safer: https://github.com/mendersoftware/gui/pull/1206/commits/78a203f06a82fd61fa43cef92b2e1e5f85375c34

**Pull request merged**